bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/HDPowerViewWebTargets.java

```diff
@@ -268,34 +335,49 @@ public void activateSceneCollection(int sceneCollectionId) throws HubProcessingE
+     * @throws HubInvalidResponseException if response is invalid
      * @throws HubProcessingException if there is any processing error
      * @throws HubMaintenanceException if the hub is down for maintenance
      */
-    public @Nullable ScheduledEvents getScheduledEvents()
-            throws JsonParseException, HubProcessingException, HubMaintenanceException {
+    public ScheduledEvents getScheduledEvents()
+            throws HubInvalidResponseException, HubProcessingException, HubMaintenanceException {
         String json = invoke(HttpMethod.GET, scheduledEvents, null, null);
-        return gson.fromJson(json, ScheduledEvents.class);
+        try {
+            ScheduledEvents scheduledEvents = gson.fromJson(json, ScheduledEvents.class);
+            if (scheduledEvents == null) {
+                throw new HubInvalidResponseException("Missing scheduledEvents response");
+            }
+            List<ScheduledEvent> scheduledEventData = scheduledEvents.scheduledEventData;
+            if (scheduledEventData == null) {
+                throw new HubInvalidResponseException("Missing 'scheduledEvents.scheduledEventData' element");
+            }
+            return scheduledEvents;
+        } catch (JsonParseException e) {
+            throw new HubInvalidResponseException("Error parsing scheduledEvents response", e);
+        }
     }
 
     /**
      * Enables or disables a scheduled event in the hub.
      * 
      * @param scheduledEventId id of the scheduled event to be enabled or disabled
      * @param enable true to enable scheduled event, false to disable
-     * @throws JsonParseException if there is a JSON parsing error
-     * @throws JsonSyntaxException if there is a JSON syntax error
+     * @throws HubInvalidResponseException if response is invalid
      * @throws HubProcessingException if there is any processing error
      * @throws HubMaintenanceException if the hub is down for maintenance
      */
     public void enableScheduledEvent(int scheduledEventId, boolean enable)
-            throws JsonParseException, HubProcessingException, HubMaintenanceException {
+            throws HubInvalidResponseException, HubProcessingException, HubMaintenanceException {
         String uri = scheduledEvents + "/" + scheduledEventId;
-        String json = invoke(HttpMethod.GET, uri, null, null);
-        JsonObject jsonObject = JsonParser.parseString(json).getAsJsonObject();
-        JsonObject scheduledEventObject = jsonObject.get("scheduledEvent").getAsJsonObject();
-        scheduledEventObject.addProperty("enabled", enable);
-        invoke(HttpMethod.PUT, uri, null, jsonObject.toString());
+        String jsonResponse = invoke(HttpMethod.GET, uri, null, null);
+        try {
+            JsonObject jsonObject = JsonParser.parseString(jsonResponse).getAsJsonObject();
+            JsonObject scheduledEventObject = jsonObject.get("scheduledEvent").getAsJsonObject();
```

Comment:
`jsonObject.get("scheduledEvent")` could in theory return null

Reply:
You are right - fixed.